src/index.js

```diff
@@ -0,0 +1,297 @@
+import ToolboxIcon from './svg/toolbox.svg';
+import './index.css';
+import Uploader from './uploader';
+
+/**
+ * Timeout when loader should be removed
+ */
+const LOADER_DELAY = 500;
+
+/**
+ * @typedef {object} PersonalityToolData
+ * @description Personality Tool's input and output data format
+ * @property {string} title — person's name
+ * @property {string} description - person's description
+ * @property {string} link - link to person's website
+ * @property {string} photo - person's photo url
+ */
+
+/**
+ * @typedef {object} PersonalityConfig
+ * @description Config supported by Tool
+ * @property {string} endpoint - image file upload url
+ * @property {string} field - field name for uploaded image
+ * @property {string} types - available mime-types
+ */
+
+/**
+ * @typedef {object} UploadResponseFormat
+ * @description This format expected from backend on file uploading
+ * @property {number} success - 1 for successful uploading, 0 for failure
+ * @property {object} file - Object with file data.
+ *                           'url' is required,
+ *                           also can contain any additional data that will be saved and passed back
+ * @property {string} file.url - [Required] image source URL
+ */
+
+/**
+ * Personality Tool for the Editor.js
+ */
+export default class Personality {
+  /**
+   * @param {PersonalityToolData} data - Tool's data
+   * @param {PersonalityConfig} config - Tool's config
+   * @param {object} api - Editor.js API
```

Comment:
```suggestion
   * @param {API} api - Editor.js API
```

![](https://capella.pics/ca61bb5a-6908-4391-8742-f9d08e4a0fd4.jpg/cover/eff252)

надо бы разобраться, как правильно грузит типы из самого редактора, чтобы уже все подсвечивалось. возможно, для этого нужно добавить editor.js в зависимости.

@gohabereg че как?